lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2832,7 +2819,8 @@ public function createEntity($className, array $data, &$hints = [])
                 }
             }
 
-            if ($entity instanceof Proxy && ! $entity->__isInitialized()) {
+            if ($this->isUninitializedObject($entity)) {
+                assert($entity instanceof InternalProxy);
```

Comment:
A proper `@psalm-assert-if-true` annotation on the new `isUninitializedObject()` method and we should not need this `assert()`.

https://psalm.dev/docs/annotating_code/adding_assertions/

Reply:
Updated, thanks.